girder-dandi-archive/girder_dandi_archive/schema_validation.py

```diff
@@ -0,0 +1,26 @@
+from semantic_version import Version
+
+from girder.exceptions import ValidationException
+
+from .constants import SCHEMA_MINIMUM_VERSION
+
+
+def validate_schema(event):
+    meta = event.info["meta"]
+    if "dandiset" not in meta:
+        # Do not validate if there is no dandiset metadata property
+        # Only top level dandiset folders should be validated
```

Comment:
well -- client must not provide metadata for any other folder!  I would fail here to catch some bug where client tries to do unexpected

Reply:
This method is run whenever any folder is saved by any client. I use the fact that only root level dandiset folders have the `dandiset` metadata property to ensure that only those folders are validated. I wanted to document that fact here.